packages/rich-list-core/src/RichListCore.ts

```diff
@@ -0,0 +1,231 @@
+        }
+      }
+    }
+  }
+
+  async get (tokenId: string): Promise<AddressBalance[]> {
+    if (Number.isNaN(tokenId)) {
+      throw new Error('Invalid token id')
+    }
+
+    if (!(await this._listTokenIds()).includes(Number(tokenId))) {
+      throw new Error('Invalid token id')
+    }
+
+    return (await this.addressBalances.list({
+      partition: `${tokenId}`,
+      limit: this.richListLength,
+      order: 'DESC'
+    })).map(s => s.data)
+  }
+
+  /**
+   * Updated rich list with latest balance
+   * by sorting the existing rich list together with `queuedAddressLimit` number of recently active addresses
+   *
+   * @param queuedAddressLimit [5000]
+   */
+  async calculateNext (queuedAddressLimit = 5000): Promise<void> {
+    const tokens = await this._listTokenIds()
+    let updatedBalances = await this._getActiveAddressBalances(tokens, queuedAddressLimit)
+
+    while (Object.keys(updatedBalances).length > 0) {
+      for (const address of Object.keys(updatedBalances)) {
+        const accountAmount = updatedBalances[address]
+        for (const tokenId of tokens) {
+          const balance = accountAmount[tokenId].toNumber()
+          const satoshi = accountAmount[tokenId].times('1e8').dividedToIntegerBy(1).toNumber()
+          await this.addressBalances.put({
+            partition: `${tokenId}`,
+            sort: satoshi,
+            id: `${address}-${tokenId}`,
+            data: {
+              address: address,
+              amount: balance
+            }
+          })
+        }
+      }
+
+      updatedBalances = await this._getActiveAddressBalances(tokens, queuedAddressLimit)
+    }
+  }
+
+  private async _getActiveAddressBalances (tokens: number[], queuedAddressLimit: number): Promise<ActiveAddressAccountAmount> {
+    const queue = await this._addressQueue()
+    const addresses = await queue.receive(queuedAddressLimit)
+
+    const balances: Record<string, AccountAmount> = {}
+    for (const a of addresses) {
+      const nonZeroBalances = await this.apiClient.account.getTokenBalances(
+        { limit: Number.MAX_SAFE_INTEGER },
+        true
+      ) as any as AccountAmount
+      balances[a] = this._appendZeroBalances(nonZeroBalances, tokens)
+      // TBD: should be combine utxo and DFI rich list
+      balances[a]['-1'] = new BigNumber(await this.whaleApiClient.address.getBalance(a))
+    }
+    return balances
+  }
+
+  private _appendZeroBalances (tokenBalances: AccountAmount, tokens: number[]): AccountAmount {
+    const result: AccountAmount = {}
+    for (const t of tokens) {
+      result[t] = tokenBalances[t] ?? new BigNumber(0)
+    }
+    return result
+  }
+
+  private async _getCrawledTip (): Promise<Schema<CrawledBlock> | undefined> {
+    const [lastBlock] = await this.crawledBlockHashes.list({
+      partition: 'NONE',
+      order: 'DESC',
+      limit: 1
+    })
+    return lastBlock
+  }
+
+  private async _invalidateDroppedOutRichList (tokenId: string, height: number): Promise<void> {
```

Comment:
```suggestion
  private async invalidateDroppedOutRichList (tokenId: string, height: number): Promise<void> {
```

In the world of typescript, I think the use of `_` is redundant since you already declare it's private. What do you think?

Reply:
I have no preference.